src/idl_gen_cpp.cpp

```diff
@@ -716,23 +733,32 @@ class CppGenerator : public BaseGenerator {
     auto num_fields =
         struct_def ? struct_def->fields.vec.size() : enum_def->vals.vec.size();
     code_.SetValue("NUM_FIELDS", NumToString(num_fields));
+    bool has_field_attributes = false;
     std::vector<std::string> names;
     std::vector<Type> types;
-    bool consecutive_enum_from_zero = true;
+    const SymbolTable<Value> *attrs_table;
+    std::vector<const SymbolTable<Value>*> type_attrs;
+    bool consecutive_enum_from_zero = false;
     if (struct_def) {
+      attrs_table = &struct_def->attributes;
```

Comment:
both StructDef and EnumDef inherit from Definition which holds attributes, so this this code can be simplified.